sql/core/src/test/scala/org/apache/spark/sql/CollationStringExpressionsSuite.scala

```diff
@@ -70,6 +71,34 @@ class CollationStringExpressionsSuite extends QueryTest with SharedSparkSession
     })
   }
 
+  case class RepeatCollationTestCase[R](s: String, n: Int, collation: String, expectedResult: R)
+
+  test("Support Repeat string expression with Collation") {
+    // Supported collations
+    val checksRepeat = Seq(
+      RepeatCollationTestCase("Spark", 2, "UTF8_BINARY", "SparkSpark"),
+      RepeatCollationTestCase("Spark", 2, "UTF8_BINARY_LCASE", "SparkSpark"),
+      RepeatCollationTestCase("Spark", 2, "UNICODE", "SparkSpark"),
+      RepeatCollationTestCase("Spark", 2, "UNICODE_CI", "SparkSpark")
+    )
+    checksRepeat.foreach(ct => {
+      checkAnswer(sql(s"SELECT repeat(collate('${ct.s}', '${ct.collation}'), ${ct.n})"),
+        Row(ct.expectedResult))
+    })
+
+    // Check return type
+    val checksReturnType = Seq(
+      RepeatCollationTestCase("Spark", 2, "UTF8_BINARY", "UTF8_BINARY"),
+      RepeatCollationTestCase("Spark", 2, "UTF8_BINARY_LCASE", "UTF8_BINARY_LCASE"),
+      RepeatCollationTestCase("Spark", 2, "UNICODE", "UNICODE"),
+      RepeatCollationTestCase("Spark", 2, "UNICODE_CI", "UNICODE_CI")
+    )
+    checksReturnType.foreach(ct => {
+      checkAnswer(sql(s"SELECT collation(repeat(collate('${ct.s}', '${ct.collation}'), ${ct.n}))"),
```

Comment:
Similar to @cloud-fan comment at https://github.com/apache/spark/pull/45611#discussion_r1533994036. Let's write unit tests, and don't test entire stack by running end-to-end tests.

Reply:
I have changed the tests now. Please check it again